c/libeulabeia/include/eulabeia/types.h

```diff
@@ -0,0 +1,272 @@
+#ifndef __EULABEIA_DATA_H
+#define __EULABEIA_DATA_H
+/**
+ * @brief defines the scan result states.
+ *
+ * The first parameter is used to generate enum eulabeia_scan_result_state and
+ * the second parameter is used to translate from and to the json
+ * representation.
+ *
+ * To add a new state add it here and NOT within enum
+ * eulabeia_scan_result_state.
+ *
+ */
+#define EULABEIA_SCAN_RESULT_STATES                                            \
+	X(EULABEIA_SCAN_RESULT_REQUESTED, requested)                           \
+	X(EULABEIA_SCAN_RESULT_QUEUED, queued)                                 \
+	X(EULABEIA_SCAN_RESULT_INIT, init)                                     \
+	X(EULABEIA_SCAN_RESULT_RUNNING, running)                               \
+	X(EULABEIA_SCAN_RESULT_STOPPING, stopping)                             \
+	X(EULABEIA_SCAN_RESULT_STOPPED, stopped)                               \
+	X(EULABEIA_SCAN_RESULT_INTERRUPTED, interrupted)                       \
```

Comment:
what is the difference in failed and interrupted?

Reply:
failed is set when there is an failed message; e.g. when the scan id is not available